src/runtime/propertyobject.cs

```diff
@@ -91,7 +96,13 @@ public static IntPtr tp_descr_get(IntPtr ds, IntPtr ob, IntPtr tp)
         public new static int tp_descr_set(IntPtr ds, IntPtr ob, IntPtr val)
         {
             var self = (PropertyObject)GetManagedObject(ds);
-            MethodInfo setter = self.setter;
+            if (!self.info.Valid)
+            {
+                Exceptions.RaiseTypeError(self.info.DeletedMessage);
+                return -1;
+            }
+
+            MethodInfo setter = self.setter.Value;
```

Comment:
should we check self.setter.Valid?

Reply:
No, but we should use UnsafeValue (doesn't throw), as null is a valid value in this context.

Reply:
No, but we should use UnsafeValue (doesn't throw), as null is a valid value in this context.